src/internal.c

```diff
@@ -16928,6 +17431,141 @@ int DoSessionTicket(WOLFSSL* ssl,
             }
             break;
         #endif /* !NO_DH && !NO_PSK */
+        #if defined(HAVE_ECC) && !defined(NO_PSK)
+            case ecdhe_psk_kea:
+            {
+                byte* pms = ssl->arrays->preMasterSecret;
+                word16 clientSz;
+
+                /* sanity check that PSK server callback has been set */
+                if (ssl->options.server_psk_cb == NULL) {
+                    WOLFSSL_MSG("No server PSK callback set");
+                    return PSK_KEY_ERROR;
+                }
+
+                /* Read in the PSK hint */
+                if ((*inOutIdx - begin) + OPAQUE16_LEN > size) {
+                    return BUFFER_ERROR;
+                }
+
+                ato16(input + *inOutIdx, &clientSz);
+                *inOutIdx += OPAQUE16_LEN;
+                if (clientSz > MAX_PSK_ID_LEN) {
+                    return CLIENT_ID_ERROR;
+                }
+
+                if ((*inOutIdx - begin) + clientSz > size) {
+                    return BUFFER_ERROR;
+                }
+
+                XMEMCPY(ssl->arrays->client_identity,
+                                                   input + *inOutIdx, clientSz);
+                *inOutIdx += clientSz;
+                ssl->arrays->client_identity[min(clientSz, MAX_PSK_ID_LEN-1)] =
+                                                                              0;
+
+                /* ECC key */
+                if ((*inOutIdx - begin) + OPAQUE8_LEN > size) {
+                    return BUFFER_ERROR;
+                }
+
+                length = input[(*inOutIdx)++];
+
+                if ((*inOutIdx - begin) + length > size) {
+                    return BUFFER_ERROR;
+                }
+
+                if (ssl->peerEccKey == NULL) {
+                    /* alloc/init on demand */
+                    ssl->peerEccKey = (ecc_key*)XMALLOC(sizeof(ecc_key),
+                                              ssl->ctx->heap, DYNAMIC_TYPE_ECC);
+                    if (ssl->peerEccKey == NULL) {
+                        WOLFSSL_MSG("PeerEccKey Memory error");
+                        return MEMORY_E;
+                    }
+                    wc_ecc_init(ssl->peerEccKey);
+                } else if (ssl->peerEccKeyPresent) {  /* don't leak on reuse */
+                    wc_ecc_free(ssl->peerEccKey);
+                    ssl->peerEccKeyPresent = 0;
+                    wc_ecc_init(ssl->peerEccKey);
+                }
+
+                if (wc_ecc_import_x963(input + *inOutIdx, length,
+                                                             ssl->peerEccKey)) {
+                    return ECC_PEERKEY_ERROR;
+                }
+
+                *inOutIdx += length;
+                ssl->peerEccKeyPresent = 1;
+
+                length = sizeof(ssl->arrays->preMasterSecret);
```

Comment:
If preMasterSecret switched to a dynamic array in the future this sizeof() calculation would be wrong.  Maybe at least a comment that includes the ENCRYPT_LEN current size, if someone changes ENCRYPT_LEN they may find it.